src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -1144,3 +1165,21 @@ def derive_column_states_from_subquery(
             state_dict=column_states,
         )
     return column_states
+
+
+def derive_data_generator_exp(expressions: Optional[List["Expression"]]) -> bool:
```

Comment:
How about `has_data_generator_expressions` since this returns a boolean? Also, given data generator also includes window function, we should either rename "data_generator", or at least add a comment

Reply:
Renamed.